File: Basic Problem Solving/All divisors/divisors.py
import math
def printDivisors(num) :
    res = [1]
    for i in range(2,num+1):
        if(num%i == 0):
            res.append(i)
    return res

def optimalSol(num):
    res = []
    sqrt = int(math.sqrt(num))
    for i in range(1,sqrt+1):
        if num%i == 0:
            res.append(i)
            if i != num//i:
                res.append(num//i)
    res.sort()
    return res

print(printDivisors(36))
print(printDivisors(17))
print(optimalSol(36))
print(optimalSol(17))

File: Basic Problem Solving/All divisors/divisors.js
const printDivisors = (num) => {
    let res = [1]
    for(let i = 2; i <= num;i++){
        if(num%i==0){
            res.push(i)
        }
    }
    return res
}

const optimalSol = (num) => {
    let res = []
    let sqrt = Math.floor(Math.sqrt(num))

    for(let i = 1; i <= sqrt; i++){
        if(num%i==0){
            res.push(i)
            if(i != Math.floor(num/i)){
                res.push(Math.floor(num/i))
            }
        }
    }
    return res.sort((a,b)=> a - b)
}


console.log(printDivisors(36))
console.log(printDivisors(17))
console.log(optimalSol(36))
console.log(optimalSol(17))
